BUG: `np.unique_values(...)` isn't equivalent to `np.unique(..., equal_nan=False)` on `main`

### Describe the issue:

`unique_values` [is documented](https://numpy.org/doc/stable/reference/generated/numpy.unique_values.html#numpy-unique-values) as an alternative to `unique` with `equal_nan=False`.
A test in `unyt`, for `np.unique_values`, was written within the assumption that the outputs are expected to be strictly equal, but lately, [it hasn't been the case](https://github.com/yt-project/unyt/issues/564). Was this assumption incorrect, or is this a regression in numpy ?

I can reproduce the error without unyt, so I know it's not a bug on our side.
Further more, it could bisect the change to https://github.com/numpy/numpy/commit/9e557eb0b621bbb92c4453b9674bb818c587845e (gh-26018)

### Reproduce the code example:

```python
# t.py
import numpy as np
from numpy.testing import assert_array_equal


def test_unique_value():
    arr = np.arange(9).reshape(3, 3)
    expected = np.unique(arr, equal_nan=False)
    out = np.unique_values(arr)
    assert_array_equal(out, expected)
```

### Error message:

```shell
❯ pytest t.py                    
============================ test session starts =============================
platform darwin -- Python 3.13.2, pytest-8.3.5, pluggy-1.5.0
rootdir: /Users/clm/dev/yt-project/unyt
configfile: pyproject.toml
plugins: cov-6.0.0, doctestplus-1.4.0
collected 1 item                                                             

t.py F                                                          [100%]

================================== FAILURES ==================================
_____________________________ test_unique_value ______________________________

    def test_unique_value():
        arr = np.arange(9).reshape(3,3)
        expected = np.unique(arr, equal_nan=False)
        out = np.unique_values(arr)
>       assert_array_equal(out, expected)
E       AssertionError: 
E       Arrays are not equal
E       
E       Mismatched elements: 8 / 9 (88.9%)
E       Max absolute difference among violations: 8
E       Max relative difference among violations: 6.
E        ACTUAL: array([8, 7, 6, 5, 4, 3, 2, 1, 0])
E        DESIRED: array([0, 1, 2, 3, 4, 5, 6, 7, 8])

t.py:8: AssertionError
========================== short test summary info ===========================
FAILED t.py::test_unique_value - AssertionError: 
============================= 1 failed in 0.03s ==============================
```

### Python and NumPy Versions:

```
2.3.0.dev0+git20250227.9e557eb
3.13.2 (main, Mar 11 2025, 17:30:09) [Clang 20.1.0 ]
```

### Runtime Environment:

```
[{'numpy_version': '2.3.0.dev0+git20250227.9e557eb',
  'python': '3.13.2 (main, Mar 11 2025, 17:30:09) [Clang 20.1.0 ]',
  'uname': uname_result(system='Darwin', node='wifibridge.home', release='24.3.0', version='Darwin Kernel Version 24.3.0: Thu Jan  2 20:24:23 PST 2025; root:xnu-11215.81.4~3/RELEASE_ARM64_T6020', machine='arm64')},
 {'simd_extensions': {'baseline': ['NEON', 'NEON_FP16', 'NEON_VFPV4', 'ASIMD'],
                      'found': ['ASIMDHP'],
                      'not_found': ['ASIMDFHM']}}]
```
### Context for the issue:

_No response_

--- a/numpy/lib/_arraysetops_impl.py
+++ b/numpy/lib/_arraysetops_impl.py
@@ -612,7 +612,11 @@ def unique_values(x):
 
     This function is an Array API compatible alternative to::
 
-        np.unique(x, equal_nan=False)
+        np.unique(x, equal_nan=False, sorted=False)
+
+    .. versionchanged:: 2.3
+       The algorithm was changed to a faster one that does not rely on
+       sorting, and hence the results are no longer implicitly sorted.
 
     Parameters
     ----------

